src/python/pants/engine/target.py

```diff
@@ -1520,6 +1526,32 @@ class GeneratedSources:
     snapshot: Snapshot
 
 
+class SourcesPaths(Paths):
```

Comment:
I tried using composition instead of inheritance, and it resulted in clunky call sites.